internal/bminventory/inventory.go

```diff
@@ -249,7 +266,12 @@ func (b *bareMetalInventory) DeregisterCluster(ctx context.Context, params inven
 }
 
 func (b *bareMetalInventory) DownloadClusterISO(ctx context.Context, params inventory.DownloadClusterISOParams) middleware.Responder {
-	if err := b.createImageJob(ctx, params.ClusterID); err != nil {
+	var cluster *models.Cluster
```

Comment:
here it was find. you have an object and you use its pointer. 
```suggestion
	var cluster models.Cluster
	if err := b.db.First(&cluster, "id = ?", params.ClusterID); err != nil {
```